### ISSUE TITLE
aws_dlm_lifecycle_policy: support other time intervals

### ISSUE THREAD

**Issue by djmetzle** [terraform-providers/terraform-provider-aws#14832]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The `aws_dlm_lifecycle_policy` resource only supports the `HOURS` interval specficiation.

https://github.com/terraform-providers/terraform-provider-aws/blob/11301c1356f6e76b07e895c0b41a326a0fabc080/aws/resource_aws_dlm_lifecycle_policy.go#L75-L82

See the docs here:
https://github.com/terraform-providers/terraform-provider-aws/blob/1d896f5b6e3ec9cbb8aedeb947b8cec275e3f25f/website/docs/r/dlm_lifecycle_policy.markdown#create-rule-arguments

It mentions:
> interval_unit - (Optional) The unit for how often the lifecycle policy should be evaluated. HOURS is currently the only allowed value and also the default value.

The go sdk does support other intervals though:

https://github.com/aws/aws-sdk-go/blob/fde575c64841b291899bc112dfcdc206f609a305/service/dlm/api.go#L2505-L2517

Along with the only currently allowed value:
https://github.com/aws/aws-sdk-go/blob/fde575c64841b291899bc112dfcdc206f609a305/service/dlm/api.go#L2465-L2468

---

It would be useful to fully use the capabilities of DLM from this Terrafrom module. 

Unfortunately, the AWS docs do not mention the other schedule intervals, so perhaps i'm misreading here:
https://docs.aws.amazon.com/dlm/latest/APIReference/API_CreateRule.html#dlm-Type-CreateRule-IntervalUnit

### New or Affected Resource(s)

* aws_dlm_lifecycle_policy

**Comment by anGie44** [comment]
Hi @djmetzle, thank you for raising this feature request. At this time, it seems the `creation rule` via the `create_rule` configuration block still only supports the `HOURS` intervalUnit but the `retain rule` on the other hand does support additional interval Units as you've referenced in the API specs (also available here: https://docs.aws.amazon.com/dlm/latest/APIReference/API_RetainRule.html) . the `aws_dlm_lifecycle_policy` resource does not currently have configurable `interval` and `interval_unit` values , so if that is still along the lines of this request, we can definitely support those changes.

**Comment by denzhel** [comment]
Any update on this ? 

We really want to use this resource to create a DLM with 3 different policies:
* Daily
* Weekly
* Monthly

The limitation of `hours` and maximum of `24` is VERY limiting and forces us to use scripts or other hacks.

Will you improve the usability any time soon ? @anGie44 @djmetzle

**Comment by danielbeardsley** [comment]
> At this time, it seems the `creation rule` via the `create_rule` configuration block still only supports the `HOURS` intervalUnit

True, but it looks like AWS now supports arbitrary schedules via the `CronExpression` rule:
https://docs.aws.amazon.com/dlm/latest/APIReference/API_CreateRule.html

See issue #13627
